src/HttpAsyncClient.php

```diff
@@ -19,7 +19,7 @@
      *
      * @param RequestInterface $request
      *
-     * @return Promise
+     * @return Promise MUST return with a PSR-7 Response.
```

Comment:
Not a fan :p

More like : 

The promise resolve a ResponseInterface or fail with a Http\Client\Exception